<launch>
<!-- Launch file for Livox MID360 LiDAR -->

	<arg name="rviz" default="false" />

	<rosparam command="load" file="$(find fast_lio)/config/mid360.yaml" />

	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="2"/>
	<param name="max_iteration" type="int" value="3" />
	<param name="filter_size_surf" type="double" value="0.5" />
	<param name="filter_size_map" type="double" value="0.5" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" /> 
    
    <!-- box_clipping arg ,初始时刻是10 Created by pf -->
    <param name="length_x_clipped" type="double" value="10.0" />
    <param name="length_y_clipped" type="double" value="10.0" />
    <param name="length_z_clipped" type="double" value="0.4" />
<!-- 设置障碍物的大小 Created by pf-->
    <param name="lidar_obstacle_radius" type="double" value="4" />
<!--图像的大小 Created by pf -->
    <param name="IMG_WIDTH" type="int" value="1000" />
    <param name="IMG_HEIGHT" type="int" value="1000" />
	<!-- <node name="pointCloud_clipping" pkg="fast_lio"  type="pointCloud_clipping"/> -->

	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
	</group>

</launch>
